steps/measure.py

```diff
@@ -139,15 +152,53 @@ def hamiltonian_analysis(
     else:
         expecval = None
 
-    K_coeff, nterms, frame_meas = estimate_nmeas_for_operator(operator, decomposition_method, expecval)
-    save_nmeas_estimate(nmeas=K_coeff, nterms=nterms, frame_meas=frame_meas, filename='hamiltonian_analysis.json')
-   
+    K_coeff, nterms, frame_meas = estimate_nmeas_for_operator(
+        operator, decomposition_method, expecval
+    )
+    save_nmeas_estimate(
+        nmeas=K_coeff,
+        nterms=nterms,
+        frame_meas=frame_meas,
+        filename="hamiltonian_analysis.json",
+    )
+
+
+def grouped_hamiltonian_analysis(
+    groups: str, expectation_values: str = "None",
+):
+
+    grouped_operator = load_qubit_operator_set(groups)
+
+    if expectation_values != "None":
+        expecval = load_expectation_values(expectation_values)
+    else:
+        expecval = None
+
+    K_coeff, nterms, frame_meas = estimate_nmeas_for_frames(grouped_operator, expecval)
+
+    save_nmeas_estimate(
+        nmeas=K_coeff,
+        nterms=nterms,
+        frame_meas=frame_meas,
+        filename="hamiltonian_analysis.json",
+    )
+
+
 def expectation_values_from_rdms(
-    interactionrdm: str,
-    qubit_operator: str,
-    sort_terms: bool = False,
+    interactionrdm: str, qubit_operator: str, sort_terms: bool = False,
 ):
     operator = load_qubit_operator(qubit_operator)
     rdms = load_interaction_rdm(interactionrdm)
     expecval = get_expectation_values_from_rdms(rdms, operator, sort_terms=sort_terms)
-    save_expectation_values(expecval, 'expectation_values.json')
+    save_expectation_values(expecval, "expectation_values.json")
+
+
+def expectation_values_from_rdms_for_qubitoperator_list(
```

Comment:
Docstring pls!

Reply:
Addressed in 0ff7e3c086d6027c44ee79a7c917c44d3a7fdb48